pkgs/tools/security/doas-sudo-shim/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, runCommand
+, asciidoctor
+, coreutils
+, gawk
+, glibc
+, util-linux
+, bash
+, gnugrep
+, makeBinaryWrapper
+, testers
+, doas-sudo-shim
+}:
+
+stdenv.mkDerivation rec {
+  pname = "doas-sudo-shim";
+  version = "0.1.1";
+
+  src = fetchFromGitHub {
+    owner = "jirutka";
+    repo = "doas-sudo-shim";
+    rev = "v${version}";
+    sha256 = "QYVqGxeWC7Tiz8aNY/LukwG4EW0km/RunGEfkzY/A38=";
+  };
+
+  nativeBuildInputs = [ asciidoctor makeBinaryWrapper ];
+  buildInputs = [ bash coreutils gawk glibc util-linux ];
+  checkInputs = [ gnugrep ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installFlags = [ "DESTDIR=$(out)" "PREFIX=\"\"" ];
+
+  postInstall = ''
+      wrapProgram $out/bin/sudo \
+        --prefix PATH : ${lib.makeBinPath [ bash coreutils gawk glibc util-linux ]}
+    '';
+
+  passthru.tests = {
+    helpTest = runCommand "${pname}-helpTest" {} ''
+      ${doas-sudo-shim}/bin/sudo -h > $out
+      grep -q "Execute a command as another user using doas(1)" $out
+    '';
```

Comment:
Doesn't this require some suid binary?

Reply:
Not the help of this command, since it's just prints out text
https://github.com/jirutka/doas-sudo-shim/blob/master/sudo#L4

Since due to lack of `suid`, I can't test the main functionality, this is kind of a version test. Unfortunately, it doesn't contain version information, so I just check for description to make sure that script runs, and it's the correct one. Maybe this is excessive.

Reply:
> Since due to lack of suid, I can't test the main functionality

We can do this in a nixos vm with security.wrappers.

Reply:
Yes, plan is to do it in separate PR according to comments above.
> So one way I can do it is next. I can remove this test from this PR. Then make a new PR adding a NixOS submodule to `security.doas`, like `qemu` in `libvirtd`. Then add a NixOS module test, which I would link back to package via [20.7.5](https://nixos.org/manual/nixpkgs/stable/#ssec-nixos-tests-linking). Which should fit all the standing conventions as far as I know.